src/theory/strings/infer_proof_cons.cpp

```diff
@@ -899,28 +900,39 @@ void InferProofCons::convert(InferenceId infer,
       }
       // connect via transitivity
       Node curr = eqs[0];
+      std::vector<Node> subs;
       for (size_t i = 1, esize = eqs.size(); i < esize; i++)
       {
         Node prev = curr;
-        curr = convertTrans(curr, eqs[1], psb);
+        curr = convertTrans(curr, eqs[i], psb);
         if (curr.isNull())
         {
-          break;
+          curr = prev;
+          if (!eqs[i][1].isConst()
+              && eqs[i][1].getKind() != Kind::STRING_CONCAT)
+          {
+            subs.push_back(eqs[i][1].eqNode(eqs[i][0]));
+          }
+          else
+          {
+            subs.push_back(eqs[i]);
+          }
```

Comment:
```suggestion
          subs.push_back(!eqs[i][1].isConst()
                                 && eqs[i][1].getKind() != Kind::STRING_CONCAT
                             ? eqs[i][1].eqNode(eqs[i][0])
                             : eqs[i]);
```
Also, can you add an insight of what is going on here?